src/backend/document.cpp

```diff
@@ -31,6 +31,18 @@ Document::Document(const Glib::RefPtr<Gio::File>& sourceFile)
     m_filesData.emplace_back(std::move(fileData));
 }
 
+Document::Document(const std::vector<Glib::RefPtr<Gio::File>> sourceFiles) : m_pages{Gio::ListStore<Page>::create()}
```

Comment:
I'm pretty sure it's possible to implement this constructor by first calling the already existing constructor with the first file of the `vector` and then calling `addFiles` with the remaining files. It shouldn't be necessary to modify `loadPages`.